<launch>

	<node pkg="final_challenge" type="TrajectoryLoader.py" name="Trajectory_loader">
		<param name="trajectory_topic" value="/trajectory/current"/>
	</node>

	<arg name="lookahead" default="1.5"/>
	<arg name="max_reacquire" default="3.0"/>
	<arg name="speed" default="1.5"/>

	<arg name="odom_topic" default="/pf/pose/odom"/>
	
	<node pkg="ta_lab6" type="pure_pursuit.py" name="Trajectory_follower">
		<param name="odom_topic" value="$(arg odom_topic)"/>
		
		<param name="trajectory_topic" value="/trajectory/current"/>
		<param name="lookahead" value="$(arg lookahead)"/>
		<!-- if the car is between lookahead and max_reacquire meters, it will attempt to
		     navigate to the nearest point on the trajectory rather than just stopping -->
		<param name="max_reacquire" value="$(arg max_reacquire)"/>
		<param name="speed" value="$(arg speed)"/>

		<param name="wheelbase" value="0.25"/>
		<param name="wrap" value="0"/>
		<param name="drive_topic" value="/vesc/high_level/ackermann_cmd_mux/input/nav_0"/>
	</node>
</launch>